tests/core/variant/test_array.h

```diff
@@ -97,6 +97,29 @@ TEST_CASE("[Array] append_array()") {
 	CHECK(int(arr1[1]) == 2);
 }
 
+TEST_CASE("[Array] append_array(), erase(), and insert()") {
+	Array arr1;
+	Array arr2;
+	arr1.push_back(1);
+	arr1.push_back(2);
+	arr2.push_back(3);
+	arr2.push_back(4);
+	CHECK(arr1.size() == 2);
+	arr1.append_array(arr2);
+	CHECK(arr1.size() == 4);
+	CHECK(int(arr1[2]) == 3);
+	CHECK(int(arr1[3]) == 4);
+	arr1.erase(2);
```

Comment:
`erase` doesn't remove the element at index `2`, that's `remove_at`, this erases all occurrences of `2`

Reply:
Changed the test case to use `remove_at` in place of `erase`.  Additionally changed test case title to reflect the functions used.